backends/graphs/README.md

```diff
@@ -0,0 +1,34 @@
+# Graphs Backend
+
+This backend produces visual representations of a P4 program as dot files. For
+now it only supports the generation of graphs for top-level control blocks.
+
+## Dependencies
+
+In addition to other p4c dependencies, this backend requires the Boost graph
+headers. On a Debian system, they can be installed with `sudo apt-get install
+libboost-graph-dev`.
+
+## Usage
+
+```
+mkdir out
+p4c-graphs <prog.p4> --graphs-dir out
+cd out
+dot <name>.dot -Tpng > <name>.png
+```
+
+## Improvement ideas
+
+  - use different shapes for vertices of different types (e.g. rectangles for
```

Comment:
I would highly recommend using rectangles instead of ellipses, the use the space much better. The output graphs will be much more compact.